### DISCUSSION TITLE
Run a task after all other tasks are completed.

### DISCUSSION THREAD

**Discussion by nbyavuz** [cirruslabs/cirrus-ci-docs#1180] [Ideas]
Hi,

I want to run a task after all other tasks are completed but I don't want it to depend on any other tasks. My use case is running cleaning script at the end. It doesn't depend on any other tasks but it should run after all other tasks are completed. I searched docs and discussions but I couldn't find a way to do that. Is there a way to do that already or does it makes sense to add such a feature?

Thanks!

**Comment by fkorotkov** [comment]
At the moment Cirrus only has [an option to change execution behavior of instructions](https://cirrus-ci.org/guide/writing-tasks/#execution-behavior-of-instructions) within a single task... 

But it seems you want to run a dedicated task in a dedicated instance after all the tasks finish? Do you want to run it only on a success of all the other tasks or in case of any failures as well? Do you want to re-run the cleaning task in case another task is getting re-run?

Also what kind of cleanup are you looking to do? Some API calls to clean some things? Maybe `on_build_failed` / `on_build_completed` [hooks](https://cirrus-ci.org/guide/programming-tasks/#hooks) can help. 🤔
